src/Native/Unix/System.Native/pal_random.cpp

```diff
@@ -96,9 +76,9 @@ extern "C" void SystemNative_GetNonCryptographicallySecureRandomBytes(uint8_t* b
     }
 
     // always xor srand48 over the whole buffer to get some randomness
-    // in case /dev/random is not really random
+    // in case /dev/urandom is not really random
 
-    for(i = 0; i < bufferLength; i++)
+    for (int i = 0; i < bufferLength; i++)
```

Comment:
I'm still not clear why this xor'ing is necessary, why the data we get back from urandom is insufficient.

Reply:
This was done back in the Rotor/Silveright days. At that time, we have seen cases where the random device was just returning non-random data on some systems under certain circumstances due to OS bugs. This was mitigating them.

Reply:
@stephentoub, @jkotas what do you think about removing it then?

Reply:
I would need to add fail fast if the read failed for some obscure reason in that case.

Reply:
If the function is changed to return error and all places are changed to do something about the error, I would be ok with removing it.

Reply:
I'll keep it in. The perf difference between having and not having it is just 2%